src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -227,7 +228,7 @@ private static bool AnalysisEnabled(Document document)
         {
             // change it to check active file (or visible files), not open files if active file tracking is enabled.
             // otherwise, use open file.
-            return document.IsOpen();
+            return document.IsOpen() && document.SupportDiagnostics();
```

Comment:
grammatically, this should probably be "SupportsDiagnostics" (i.e. an extra 's' in 'Supports').

Reply:
sure